src/core/server/saved_objects/migrationsv2/model.ts

```diff
@@ -181,7 +181,7 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   // control state using:
   // `const res = resW as ResponseType<typeof stateP.controlState>;`
 
-  let stateP: State = cloneDeep(currentState);
+  let stateP: State = currentState;
```

Comment:
Are we sure our state workflow is pure enough for this to not have any side effect? (should be but still asking)

Reply:
> Are we sure our state workflow is pure enough for this to not have any side effect? (should be but still asking)

@pgayvallet  I'm not quite sure `cloneDeep` usage is justified here. 
Mutations made during `model` execution will be returned from the `model` and State will be cloned during the next `model` call again. 
The only difference is the `OldState` mutation won't affect `NewState` shape. But AFAIK `SavedObjectMigrationFn` is the only valid case for mutations and we already clone the `doc` during `migrateAndConvert` call.https://github.com/elastic/kibana/blob/2f6a3ce19bab370ecac762b571aae667aa6c8eed/src/core/server/saved_objects/migrations/core/document_migrator.ts#L245
I'm fine to roll back the change. In this case, we will pay the penalty of deep cloning all the SO several times. Plus additional work for GC.

> We are risking that any step might call stateP.logs.push() 

@afharo  Can `stateP.logs.push` be called with the current implementation? I think so. `stateP.logs` is not an immutable array.